erts/emulator/nifs/win32/win_prim_file.c

```diff
@@ -795,74 +858,41 @@ posix_errno_t efile_read_info(const efile_path_t *path, int follow_links, efile_
             return efile_read_info(&resolved_path, 0, result);
         }
 
-        handle = CreateFileW((const WCHAR*)path->data, GENERIC_READ,
-            FILE_SHARE_FLAGS, NULL, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS,
-            NULL);
-
-        /* The old driver never cared whether this succeeded. */
-        if(handle != INVALID_HANDLE_VALUE) {
-            GetFileInformationByHandle(handle, &native_file_info);
-            CloseHandle(handle);
-        }
-
-        FILETIME_TO_EPOCH(result->m_time, native_file_info.ftLastWriteTime);
-        FILETIME_TO_EPOCH(result->a_time, native_file_info.ftLastAccessTime);
-        FILETIME_TO_EPOCH(result->c_time, native_file_info.ftCreationTime);
+        GetFileInformationByHandle(handle, &native_file_info);
+        CloseHandle(handle);
 
-        if(result->m_time == -EPOCH_DIFFERENCE) {
-            /* Default to 1970 just like the old driver. */
-            result->m_time = 0;
-        }
+        build_file_info_times(&native_file_info, result);
+    }
 
-        if(result->a_time == -EPOCH_DIFFERENCE) {
-            result->a_time = result->m_time;
-        }
+    build_file_info(&native_file_info, &path, is_link, result);
```

Comment:
`&path` should be `path` here.
```suggestion
    build_file_info(&native_file_info, path, is_link, result);
```